src/GafferImageUI/ImageView.cpp

```diff
@@ -275,8 +283,863 @@ GAFFER_NODE_DEFINE_TYPE( V2fContextVariable )
+		{
+			if( event.sourceGadget != this )
+			{
+				return false;
+			}
+
+			updateDragPoint( event );
+			return true;
+		}
+
+		bool dragMove( const GafferUI::DragDropEvent &event )
+		{
+			updateDragPoint( event );
+			return true;
+		}
+
+		bool dragEnd( const GafferUI::DragDropEvent &event )
+		{
+			updateDragPoint( event );
+			m_deleting = false;
+			return true;
+		}
+
+		void updateDragPoint( const GafferUI::DragDropEvent &event )
+		{
+			if( m_deleting )
+			{
+				return;
+			}
+			const V2f p = eventPosition( event );
+			V2i point = m_dragStartPlugValue;
+
+			const V2f offset = p - m_dragStart;
+			point += V2i( round( offset.x ), round( offset.y ) );
+
+			m_plug->setValue( point );
+		}
+
+		bool buttonRelease( const GafferUI::ButtonEvent &event )
+		{
+			const V2f p = eventPosition( event );
+			if( m_deleting && onDeleteButton( p ) )
+			{
+				m_deleteClickedSignal( m_plug.get() );
+			}
+			m_deleting = false;
+
+			return true;
+		}
+
+		void leave()
+		{
+			Pointer::setCurrent( "" );
+			m_hover = 0;
+			m_deleting = false;
+
+			Metadata::registerValue( m_plug.get(), g_hoveredKey, new IECore::BoolData( false ), false );
+		}
+
+		V2f viewportPlanarScale() const
+		{
+			// It's kinda silly to have to reverse engineer viewportGadget->m_planarScale because it isn't public
+			const ViewportGadget *viewportGadget = ancestor<ViewportGadget>();
+			return V2f(
+				viewportGadget->getCamera()->getAperture()[0] / viewportGadget->getViewport()[0],
+				viewportGadget->getCamera()->getAperture()[1] / viewportGadget->getViewport()[1]
+			);
+		}
+
+		bool onDeleteButton( const V2f &p ) const
+		{
+			// Any positions that are part of the gadget, but not part of an extended bound, are
+			// on the delete button
+			V2f point = V2f(m_plug->getValue()) + V2f( 0.5 );
+			const V2f planarScale = viewportPlanarScale();
+			const V2f threshold( planarScale * m_handleSize );
+			return
+				p.x > point.x + 0.5 * threshold.x - planarScale.x &&
+				p.y > point.y + 0.5 * threshold.y - planarScale.y;
+		}
+
+		V2f eventPosition( const ButtonEvent &event ) const
+		{
+			const ViewportGadget *viewportGadget = ancestor<ViewportGadget>();
+			const ImageGadget *imageGadget = static_cast<const ImageGadget *>( viewportGadget->getPrimaryChild() );
+			V2f pixel = imageGadget->pixelAt( event.line );
+			Context::Scope contextScope( imageGadget->getContext() );
+			pixel.x *= imageGadget->getImage()->format().getPixelAspect();
+			return pixel;
+		}
+
+		V2iPlugPtr m_plug;
+		const std::string m_id;
+		bool m_editable;
+		float m_handleSize;
+		int m_hover;
+		bool m_deleting;
+
+		DeleteClickedSignal m_deleteClickedSignal;
+
+		Imath::V2i m_dragStartPlugValue;
+		Imath::V2f m_dragStart;
+};
+GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( V2iGadget )
+
 } // namespace
 
+ImageView::ColorInspectorPlug::ColorInspectorPlug( const std::string &name )
+	: ValuePlug( name, ValuePlug::Direction::In, Default )
+{
+	addChild( new IntPlug( "mode" ) );
```

Comment:
Would be good to have min/max limits here. Reading through the code, I also found myself missing an enum to give meaning to values.